thavalon-server/src/connections/validation.rs

```diff
@@ -0,0 +1,257 @@
+        return Err(ValidationError::InvalidPassword);
+    }
+
+    let hash = scrypt::scrypt_simple(plaintext, &ScryptParams::recommended()).map_err(|e| {
+        log::error!("An RNG error occurred with the underlying OS.");
+        log::error!("{:?}", e);
+        ValidationError::HashError
+    });
+
+    hash
+}
+
+/// Validates a plaintext password against a given hash.
+///
+/// # Arguments
+///
+/// * `plaintext` - the plain text password to check
+/// * `hash` - Password hash in scrypt format
+///
+/// # Returns
+/// True if passwords match. False otherwise.
+pub async fn validate_password(plaintext: &String, hash: &String) -> bool {
+    let result = match scrypt::scrypt_check(plaintext, hash) {
+        Ok(_) => true,
+        Err(e) => {
+            if e == CheckError::InvalidFormat {
+                log::error!("Database hash is not in a valid scrypt format.");
+            }
+            false
+        }
+    };
+
+    result
+}
+
+/// Creates a valid JWT using a user's email.
+///
+/// # Arguments
+///
+/// * `user_email` - The user's email to create a JWT with.
+///
+/// # Returns
+///
+/// A JWTResponse with the JWT
+pub async fn create_jwt(
+    user_email: &String,
+    token_store: TokenStore,
+) -> (JWTResponse, RefreshTokenInfo) {
+    log::info!("Creating a new JWT for {}.", user_email);
+    let jwt_secret = env::var("JWT_SECRET").unwrap_or("JWT_SECRET".to_string());
+    let time = Utc::now();
+    let expiration_time = time
+        .checked_add_signed(Duration::minutes(60))
+        .expect("Failed to get expiration time.");
+    let claims = JWTClaims {
+        aud: user_email.clone(),
+        exp: expiration_time.timestamp(),
+        iat: time.timestamp(),
+        iss: "ThavalonGameServer".to_string(),
+        nbf: time.timestamp(),
+        sub: "ThavalonAuthenticatedUser".to_string(),
+    };
+
+    let token = jsonwebtoken::encode(
+        &Header::default(),
+        &claims,
+        &EncodingKey::from_secret(jwt_secret.as_bytes()),
+    )
+    .expect("Failed to generate a JWT for this claim.");
+
+    log::info!("Successfully created a JWT for {}.", user_email);
+    (
+        JWTResponse {
+            token_type: "Bearer".to_string(),
+            access_token: token,
+            expires_at: expiration_time.timestamp(),
+        },
+        create_refresh_token(user_email, token_store).await,
+    )
+}
+
+/// Validates a JWT given the token.
+///
+/// # Arguments
+///
+/// * `token` - A JWT to authenticate
+///
+/// # Returns
+///
+/// User ID on success, ValidationError on failure.
+pub async fn validate_jwt(token: &str) -> Result<String, ValidationError> {
+    log::info!("Validating received JWT");
+    let jwt_secret = env::var("JWT_SECRET").unwrap_or("JWT_SECRET".to_string());
+    let validation = Validation {
+        leeway: 60,
+        validate_exp: true,
+        validate_nbf: true,
+        iss: Some("ThavalonGameServer".to_string()),
```

Comment:
Should these strings be in shared constants?

Reply:
Fixed by moving to &'static strs in the TokenManager struct.